### PR TITLE
Update documentation for config.public_file_server.enabled [ci-skip]

### PR DESCRIPTION
**Pull request by Adrian-Hirt**
### Motivation / Background

#47137 turned on static file server by default for all environments. The documentation still mentions that it's set to `false` in the production environment, which is not the case anymore. This PR changes the docs to reflect the change in the previously mentioned pull request.

### Detail

This Pull Request changes the docs for `config.public_file_server.enabled`.

### Additional information

None :)

### Checklist

Before submitting the PR make sure the following are checked:

* [X] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [X] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [X] Tests are added or updated if you fix a bug or add a feature.
* [X] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

P.s. Sorry for forgetting to include [ci skip] in the PR title, I had to re-open the tab when writing the PR message and did not include it the second time :S

### CODE REVIEW

File: guides/source/configuring.md [hunk 1]
```diff
@@ -500,7 +500,7 @@ The default value depends on the `config.load_defaults` target version:
 
 #### `config.public_file_server.enabled`
 
-Configures Rails to serve static files from the public directory. This option defaults to `true`, but in the production environment it is set to `false` because the server software (e.g. NGINX or Apache) used to run the application should serve static files instead. If you are running or testing your app in production using WEBrick (it is not recommended to use WEBrick in production), set the option to `true`. Otherwise, you won't be able to use page caching and request for files that exist under the public directory.
+Configures whether Rails should serve static files from the public directory. This option defaults to true for all environments. If the server software (e.g. NGINX or Apache) used to run the application in the production environment should serve static files instead, set this value to false. If you are running or testing your app in production using WEBrick (it is not recommended to use WEBrick in production), set the option to true. Otherwise, you won't be able to use page caching and request for files that exist under the public directory.
```
Comment (jonathanhefner) [inline guides/source/configuring.md:503]: Good catch!  I think we could also remove the part about setting to `true` for WEBrick:

```suggestion
Configures whether Rails should serve static files from the public directory. 
Defaults to `true`.

If the server software (e.g. NGINX or Apache) should serve static files instead, 
set this value to `false`.
```

**Comment by Adrian-Hirt** [comment]
P.s. I also opened the related (but separate PR) #49851 to address the defaults in the generated files. As written there, I split the PRs into two such that you can accept either, both or none.

**Comment by jonathanhefner** [comment]
I pushed a commit to [squash the commits](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits).

Thank you, @Adrian-Hirt! :scroll:

**Comment by Adrian-Hirt** [comment]
Thank you as well!

**Pull request merged**